sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala

```diff
@@ -117,6 +117,21 @@ case class HashAggregateExec(
         // so return an empty iterator.
         Iterator.empty
       } else {
+        val filterPredicates = new mutable.HashMap[Int, GenPredicate]
+        aggregateExpressions.zipWithIndex.foreach{
+          case (ae: AggregateExpression, i) =>
+            ae.mode match {
+              case Partial | Complete =>
+                ae.filter.foreach { filterExpr =>
+                  val filterAttrs = filterExpr.references.toSeq
+                  val predicate = newPredicate(filterExpr, child.output ++ filterAttrs)
```

Comment:
`genInterpretedPredicate` instead of `newPredicate`?

btw, in the interpreter mode `doExecute()` of `FileterExec`, it seems we currently use generated code from `newPredicate` for evaluating predicates. Any reason that we cannot turn off codegen there via `CODEGEN_FACTORY_MODE`? @viirya @cloud-fan 
https://github.com/apache/spark/blob/e46e487b0831b39afa12ef9cff9b9133f111921b/sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala#L230

Reply:
The background of adding `CODEGEN_FACTORY_MODE` is to have a config for test only. It is easier for us to test interpreted, codegen paths separately.

For non test, I think we always go codegen first and fallback to interpreted if codegen fails.